### PR TITLE
Adjust labelling to actual naming of ICs

### PR DESCRIPTION
**Pull request by stfnrpplngr**
I'm not sure whether this is a bug, just observed the following.
Plotting the time-locked sources doing an ICA, results in different labelling clicking on the components' lines:
incorrect_labelling

Changing the proposed line in `_export_info()` solves the problem:
correct_labeling

### CODE REVIEW

**Comment by agramfort** [comment]
we keep getting these bug reports about channel names that don't correspond to indices.

when we write `ICA #001` it corresponds to the indices but when the channel names named ICA001 start with 1 to match the convention of EEG channels that start with 1.

is this really worth changing?

**Comment by Eric89GXL** [comment]
Given the persistent confusion I think we should change something. To me
each new report adds to evidence that we have a bad design that needs
fixing.

We could try labeling as `ICA[3]` or so instead of `ICA #003`, but I still
don't think it's very good.

I'm starting to think that it would be better for ICA to number their
channel names from zero. It creates an inconsistency with other channel
types, yes, but I think it will create less confusion than the current
implementation.

**Comment by agramfort** [comment]
agreed

I give up :)

let's call the ICA channels ICA 000, ICA 001, etc...

**Comment by Eric89GXL** [comment]
Thanks for being the final straw @stfnrpplngr, closing this as dup of #3889 where we'll continue discussion

**Comment by Eric89GXL** [comment]
Reopened because this is already along the lines of the right fix. @stfnrpplngr 3 more things are necessary to get this in:

1. Update ICA tutorials
2. Add the option to use old / bad naming style (like what's done in #1622) w/deprecation cycle to move to the new / good naming style
3. Updating any tests

Do you have time to look into these? We're happy to walk you through them if you want to learn how.

**Comment by stfnrpplngr** [comment]
If someone could walk me through, that would be great.

**Comment by Eric89GXL** [comment]
For (1), look through these to see if there is any mention of numbering differences:

* https://github.com/mne-tools/mne-python/blob/master/mne/preprocessing/ica.py#L118
* https://github.com/mne-tools/mne-python/blob/master/tutorials/plot_artifacts_correction_ica.py
* https://github.com/mne-tools/mne-python/blob/master/examples/preprocessing/plot_run_ica.py 

Also look through #3633, #2019, #2000, #1622 to see if they mention updating docs anywhere else that we could fix.

For (2), you'd add a `channel_start=None` argument to `ICA` constructor and in `__init__` do:
```
if channel_start is None:
    channel_start = 1
    warn('channel_start will default to 1 in 0.15, but change to 0 in 0.16. Set it explicitly to avoid this warning', DeprecationWarning)
if channel_start not in (0, 1):
    raise ValueError('channel_start must be 0 or 1, got %s' % (channel_start,))
```

For (3) run `nosetests mne/preprocessing/tests/test_ica.py` and ensure they pass. Run the two examples referenced above and compare the outputs to those from the [current dev build](mne-tools.github.io/dev/index.html). Push changes to your PR and see if the CIs (Travis, AppVeyor, and CircleCI) are happy.

**Comment by Eric89GXL** [comment]
@stfnrpplngr do you have time to give it a try, or should someone else take over?

File: tutorials/plot_artifacts_correction_ica.py [hunk 1]
```diff
@@ -30,8 +30,7 @@
 raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
 
 raw = mne.io.read_raw_fif(raw_fname, preload=True)
-# 1Hz high pass is often helpful for fitting ICA
-raw.filter(1., 40., n_jobs=2, fir_design='firwin')
+raw.filter(1, 40, n_jobs=2)  # 1Hz high pass is often helpful for fitting ICA
```
Comment (Eric89GXL) [inline tutorials/plot_artifacts_correction_ica.py]: revert

**Comment by stfnrpplngr** [comment]
@larsoner would you mind having a look on it again?

File: mne/preprocessing/ica.py [hunk 2]
```diff
@@ -224,7 +224,17 @@ class ICA(ContainsMixin):
...
+                 verbose=None, channel_start=None):  # noqa: D102
+
+        if channel_start is None:
+            channel_start = 1
+            warn('channel_start will default to 1 in 0.15, but change '
+                 'to 0 in 0.16. Set it explicitly to avoid this warning',
+                 DeprecationWarning)
+        if channel_start not in (0, 1):
+            raise ValueError('channel_start must be 0 or 1, got %s' %
+                             (channel_start,))
```
Comment (dengemann) [inline mne/preprocessing/ica.py]: do we actually really need a parameter for this? Or is the plan to remove it after the deprecation cycle?

File: mne/preprocessing/ica.py [hunk 3]
```diff
@@ -811,7 +822,7 @@ def _export_info(self, info, container, add_channels):
         ch_names = []
         ch_info = info['chs'] = []
         for ii in range(self.n_components_):
-            this_source = 'ICA %03d' % (ii + 1)
+            this_source = 'ICA %03d' % (ii)
```
Comment (dengemann) [inline mne/preprocessing/ica.py]: we could really just make clear that these are the indices. We have a similari issue in #4049 and should solve it consistently.

File: mne/preprocessing/ica.py (hunk 2 above)
Comment (larsoner) [inline mne/preprocessing/ica.py]: See suggestion above

https://github.com/mne-tools/mne-python/pull/4320#issuecomment-308744108

But maybe we don't need the cycle and can just make the API change

Comment (dengemann) [inline mne/preprocessing/ica.py]: I'd opt for not deprecating. We could even twitter vote ... Let's just get it right once.

Comment (larsoner) [inline mne/preprocessing/ica.py]: IIRC that simple change is what @stfnrpplngr had as the commit two months ago. :(

Sorry for the back and forth @stfnrpplngr, but can you change it just to use the index (no need for the param / deprecation I suggested before)? Other than that, the other thing I see is that `doc/whats_new.rst` needs to be updated in the `API` section saying that the ICA labeling now starts from 0.

**Comment by dengemann** [comment]
My apologies @stfnrpplngr <https://github.com/stfnrpplngr> I did not follow
closely enough so far. If you guys prefer a depracation I won't stop you.
But don't do it for me.

**Comment by larsoner** [comment]
I don't feel strongly either way, @stfnrpplngr feel free to revert to the simpler way or keep the current deprecation, up to you. Then we need `doc/whats_new.rst` update and we should be good

**Comment by stfnrpplngr** [comment]
I would like to keep it containing the deprecation warning hopefully guaranteeing less confusion on the user site. The ``doc/whats_new.rst`` has been changed.

**Comment by agramfort** [comment]
so now the plan is to have ICA 000 as first ica channel? to everything is zero indexed?

**Comment by larsoner** [comment]
> so now the plan is to have ICA 000 as first ica channel? to everything is zero indexed?

Yep

File: mne/preprocessing/ica.py [hunk 4]
```diff
@@ -238,7 +238,17 @@ class ICA(ContainsMixin):
     def __init__(self, n_components=None, max_pca_components=None,
                  n_pca_components=None, noise_cov=None, random_state=None,
                  method='fastica', fit_params=None, max_iter=200,
-                 verbose=None):  # noqa: D102
+                 verbose=None, channel_start=None):  # noqa: D102
```
Comment (larsoner) [inline mne/preprocessing/ica.py]: add `channel_start` before `verbose`, and also document it

**Comment by agramfort** [comment]
ok let's see if it's actually less confusing for everyone...

File: doc/whats_new.rst [hunk 5]
```diff
@@ -160,6 +160,9 @@ BUG
 
 API
 ~~~
+
+	  - Add ``channel_start`` parameter in :func:`mne.preprocessing.ica` to make explicit that ICA component labelling starts from 1 in 0.15, default will change to 0 in 0.16 by `Stefan Repplinger`_ and `Eric Larson`_
```
Comment (agramfort) [inline doc/whats_new.rst]: over indented

File: mne/preprocessing/ica.py (hunk 3 above)
Comment (agramfort) [inline mne/preprocessing/ica.py]: yes maybe using the # sign

so "ICA #1" is clear that it's index 0?

Comment (larsoner) [inline mne/preprocessing/ica.py]: I'd rather do `[0]` than `#1`. But earlier you [seemed happy with the current implementation](https://github.com/mne-tools/mne-python/pull/4320#issuecomment-308731710)...

**Comment by agramfort** [comment]
ok keep it simple stupid. Start naming at 0

**Comment by larsoner** [comment]
With no deprecation? That's okay with me I guess, and it sounded like @dengemann thought it was overkill, too.

@stfnrpplngr I'll close this, cherry-pick your commits (there are a couple that do what we want) and open a new PR. Thanks for the fix and iterations!

**Comment by larsoner** [comment]
> yes maybe using the # sign
>
> so "ICA #1" is clear that it's index 0?

BTW this suggestion is the opposite of how it works in ICA viz currently. "ICA #000" is index 0. Just more evidence that our current scheme is unintuitive.

**Pull request closed without merging**